src/test/java/com/google/sps/servlets/ContestServletTest.java

```diff
@@ -0,0 +1,122 @@
+    electionEntityOne.setProperty("id", "9999");
+    electionEntityOne.setProperty("name", "myElection");
+    electionEntityOne.setProperty("scope", "myScope");
+    electionEntityOne.setProperty("date", "myDate");
+    electionEntityOne.setProperty("contests", new HashSet<Long>());
+    electionEntityOne.setProperty("propositions", new HashSet<Long>());
+
+    contestEntityOne = new Entity("Contest", 1);
+    contestEntityOne.setProperty("name", "myFirstContest");
+    contestEntityOne.setProperty("candidates", candidateIdSet);
+    contestEntityOne.setProperty("description", "This contest is important.");
+
+    contestEntityTwo = new Entity("Contest", 2);
+    contestEntityTwo.setProperty("name", "mySecondContest");
+    contestEntityTwo.setProperty("candidates", new HashSet<Long>());
+    contestEntityTwo.setProperty("description", "This contest is also important.");
+
+    helper.tearDown();
+  }
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @Test
+  public void singleElection_singleContest_testDoGet() throws Exception {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    long contestId = ds.put(contestEntityOne).getId();
+
+    HashSet<Long> contestSet = (HashSet<Long>) electionEntityOne.getProperty("contests");
+    contestSet.add(contestId);
+    ds.put(electionEntityOne);
+
+    when(httpServletRequest.getParameter("electionId")).thenReturn("9999");
+    when(httpServletResponse.getWriter()).thenReturn(printWriter);
+
+    ContestsServlet contestServlet = new ContestsServlet();
+    contestServlet.doGet(httpServletRequest, httpServletResponse);
+
+    verify(printWriter)
+        .println(
+            "[{\"name\":\"myFirstContest\",\"candidates\":[1,2,3],\"description\":\"This contest is important.\"}]");
+  }
+
+  @Test
+  public void singleElection_twoContest_testDoGet() throws Exception {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    long contestOneId = ds.put(contestEntityOne).getId();
+    long contestTwoId = ds.put(contestEntityTwo).getId();
+
+    HashSet<Long> contestSet = (HashSet<Long>) electionEntityOne.getProperty("contests");
+    contestSet.add(contestOneId);
+    contestSet.add(contestTwoId);
+    ds.put(electionEntityOne);
+
+    when(httpServletRequest.getParameter("electionId")).thenReturn("9999");
+    when(httpServletResponse.getWriter()).thenReturn(printWriter);
+
+    ContestsServlet contestServlet = new ContestsServlet();
+    contestServlet.doGet(httpServletRequest, httpServletResponse);
+
+    verify(printWriter)
+        .println(
+            "[{\"name\":\"myFirstContest\",\"candidates\":[1,2,3],\"description\":\"This contest is important.\"},"
+                + "{\"name\":\"mySecondContest\",\"candidates\":[],\"description\":\"This contest is also important.\"}]");
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+    electionEntityOne.setProperty("contests", new HashSet<Long>());
```

Comment:
If you create electionentityone in the each unit test, there's no need to clean this stuff up. Which is nice because it hurts readability.